clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -224,6 +224,13 @@ boolean canAutoCommit() {
         return autoCommitState.isPresent() && !subscriptions.allConsumed().isEmpty();
     }
 
+    /**
+     * Updates the member ID and epoch upon receiving ConsumerGroupHeartbeatResponse.
+     */
+    void updateMemberInformation(String memberId, int memberEpoch) {
+        groupState.generation = new GroupState.Generation(memberEpoch, memberId, null);
+    }
```

Comment:
I wonder if this the right way to do this. Wouldn't it be better to let the CommitRequestManager get those from the MembershipManager instead of duplicating them? This issue with this approach is that you have to ensure that `updateMemberInformation` is called in all the relevant places. For instance, the member epoch is also updated when the member is fenced.

Reply:
Understood. I suppose that this is what @lianetm is working on. It makes sense as a temporary fix.

Reply:
There are two other places where the member epoch is updated. Should we also call `updateMemberInformation` for those for completeness?

Reply:
Yes, I'm on this and will come out in the commit PR, so if we want to unblock the e2e I'm ok with keeping this as temporary fix.

Reply:
Just for the record, with the fix in the commit PR the approach is having this CommitManager as a StateListener, that gets notified when epoch changes, so it gets the last value from the MembershipManager. I did the listener approach to avoid a circular dep (the MembershipMgr uses the CommitMgr to commit), and truly, this Commit does not need the whole MembershipMgr, it just needs to have the latest ID/epoch, so the listener seems appropiate. Anyways, let's have this conversation on the other PR when in review.